src/docs/arch-rts.dox

```diff
@@ -1,113 +1,114 @@
-updates are aborted and the on-disk version updates are replaced with an update
-from history store otherwise the data store version is removed.
+RTS scans tables in the database to remove any unstable modifications. A table is selected by RTS if
+it meets one of the following conditions:
 
-Rollback to stable is performed in three phases
-1. WiredTiger startup
-2. WiredTiger shutdown
-3. Application initiated
+1. It is modified.
+2. It has prepared updates.
+3. It has updates from transactions greater than the recovery checkpoint snapshot (this is
+applicable only during the restart phase)
+4. One of the checkpoint durable timestamps (start/stop) is greater than the stable timestamp.
 
-To improve the performance of rollback to stable operation, rollback to stable
-will perform only on particular tables that need rollback. Rollback to stable
-doesn't operate on logged tables as the updates on these are stable when the
-transaction gets committed.
-
-@section rts-stable-update Stable update of rollback to stable
-
-According to rollback to stable, the stable version of update is an update that
-has durable timestamp of less than or equal to the stable timestamp and it's
-transaction id must be committed according to the checkpoint snapshot.
+On the other hand, a table is skipped by RTS if it does not meet any of the previous criteria but
+does meet one of the following:
 
-@section rts-preconditions Pre-conditions required for rollback to stable
+1. It is empty
+2. It has timestamped updates but there is no stable timestamp set.
+3. The associated files are missing.
+4. The associated files are corrupted.
+5. It is dedicated to @ref arch-metadata "metadata"
+6. It is a logged table. RTS is not necessary for those as their updates are stable when the
+transaction gets committed.
+7. It has no aggregated time window. It is worth noting that the tables with no aggregate time
+window are only created in versions older than MongoDB 4.4
 
-To perform rollback to stable, there shouldn't be any transaction activity happening
-in the WiredTiger.
+For each selected table, its pages are read into the cache. The unstable in-memory updates and the
+unstable historical versions from the history store are removed. The on-disk version is replaced by
+the latest stable in-memory update if any, otherwise by the latest stable version from the history
+store. If there is no stable in-memory updates and no stable version in the history store, the data
+store entry is completely removed. In the situation where the table has no timestamped updates, all
+the historical versions related to this table are removed from the history store as they are no
+longer required and the on-disk value is preserved.
 
-@section rts-table-check Checks performed on a table by rollback to stable
+Finally, RTS is triggered on startup, shutdown or when the application is initiated. It is worth
+noting that it cannot run if there is any transaction activity happening in WiredTiger.
```

Comment:
May be RTS needs exclusive access of the database and no concurrent transaction is allowed.

Reply:
Thanks - 473c87cb788461e775cacf63f16b206fd49d92ac